crypto/fipsmodule/ml_dsa/ml_dsa.c

```diff
@@ -44,7 +45,11 @@ int ml_dsa_44_keypair(uint8_t *public_key   /* OUT */,
   boringssl_ensure_ml_dsa_self_test();
   ml_dsa_params params;
   ml_dsa_44_params_init(&params);
-  return (ml_dsa_keypair(&params, public_key, private_key, seed) == 0);
+  int ret = (ml_dsa_keypair(&params, public_key, private_key, seed) == 0);
```

Comment:
Any ML-DSA function that calls another approved service (e.g. SHA3) or has a self test needs to lock the service indicator before the call, and then unlock the service indicator after. Otherwise it's not clear to the caller if the ML-DSA call they made is approved or if it's the SHA3 call that was approved but ML-DSA was not approved. This is why the CI is failing with `Assertion failed: (before + 1 == after), function TestBody, file service_indicator_test.cc, line 5413.`. 

https://github.com/aws/aws-lc/pull/2112 has more information on the issue and examples.